src/FatLib/FatFile.cpp

```diff
@@ -1498,22 +1498,38 @@ size_t FatFile::write(const void* buf, size_t nbyte) {
 //-------------------------------------------------------
 // EFP3 - Match upstream Arduino hack
 int FatFile::availableSpaceForWrite() {
-  // error if not a normal file or is read-only
-  if (!isWritable() || m_vol->m_blockDev->isBusy()) {
+  if (!isWritable()) {
+    // error if not a normal file or is read-only
     return 0;
   }
+
+  if (m_vol->m_blockDev->isBusy()) {
+    // SdSpiCard::isTimedOut is private...
+    decltype(millis()) t0 = millis();
+    while (m_vol->m_blockDev->isBusy()) {
+      if ((millis() - t0) > SD_READ_TIMEOUT) {
```

Comment:
@d-a-v, I don't believe there is any aynch processing in the SdFat library.  Given that, I'm not sure how this can ever do anything but delay a while and return the same 0...

Reply:
I have tested this and it has an effect.
Just after an operation (a write operation I think) this function returns 0 because of `isBusy()` and with this change, it returns the right value. 
Maybe the SDCard tells that it's not the good moment, given [``isBusy()``'s code](
https://github.com/earlephilhower/ESP8266SdFat/blob/3f50e5547d20854d4dfdc715446758ccc734bec2/src/SdCard/SdSpiCard.cpp#L438-L454).

- I can provide a snippet to reproduce
- I can remove this part of the patch because it is OK if `afw()` returns 0 sometimes and `afw()` is not really meant to delay.